src/android/shakeDetector.java

```diff
@@ -0,0 +1,258 @@
+                float zAcceleration = event.values[2];
+
+                //Used to diminish the effect of the acceleration in the y-direction
+                float yAccelerationConst = 0.6f;
+
+                lastAcceleration = currAcceleration;
+                currAcceleration = (float) Math.sqrt((double) (zAcceleration * zAcceleration + yAcceleration * yAcceleration * yAccelerationConst * yAccelerationConst));
+
+                acceleration = acceleration * 0.95f + (currAcceleration - lastAcceleration);
+                String AccTag1 = "2Shake0.6A0.95-13-10";
+
+                Log.i(this, AccTag1, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration) + ",");
+
+                /*
+                if ((acceleration > 13f) && (Math.abs(xAcceleration) < 10)) {
+                    lastUpdated = currTime;
+                    NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, AccTag1);
+                    //Log.i(this, AccTag1, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration) + ",");
+                    notifyEvent(this, AccTag1, new JSONObject());
+		        }
+                */
+
+                acceleration2 = acceleration2 * 0.85f + (currAcceleration - lastAcceleration);
+                String AccTag2 = "2Shake0.6A0.85-13-10";
+
+                Log.i(this, AccTag2, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
+
+/*
+                if ((acceleration > 13f) && (Math.abs(xAcceleration) < 10)) {
+                    lastUpdated = currTime;
+                    NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, AccTag2);
+                    //Log.i(this, AccTag2, ","+String.valueOf(currTime)+"," + Float.toString(xAcceleration) + "," + Float.toString(yAcceleration) + "," + Float.toString(zAcceleration) + "," + Float.toString(acceleration2) + ",");
+                }*/
+                notifyEvent(this, AccTag2, new JSONObject());
+            }
+            else if(event.sensor.getType() == Sensor.TYPE_GYROSCOPE){
+                float xRotation = event.values[0];
+                String GyroTag = "2Gyro5";
+
+                Log.i(this, GyroTag, ","+String.valueOf(currTime)+"," + Float.toString(xRotation) + "," );
+
+                /*
+                if (xRotation > 5f) {
+                    lastUpdated = currTime;
+                    NotificationHelper.createNotification(this, Constants.TRACKING_ERROR_ID, GyroTag);
+                    //Log.i(this, GyroTag, "," + Float.toString(xRotation) + "," );
+                }*/
+                notifyEvent(this, GyroTag, new JSONObject());
```

Comment:
if your eventName tags are things like `2Gyro5`, you should probably put them into the javascript so that people can configure listeners for them